test/Microsoft.TemplateEngine.Cli.UnitTests/ParserTests/TabCompletionTests.cs

```diff
@@ -0,0 +1,317 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using FakeItEasy;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Cli.Commands;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Edge.Settings;
+using Microsoft.TemplateEngine.Mocks;
+using Microsoft.TemplateEngine.TestHelper;
+using Xunit;
+
+namespace Microsoft.TemplateEngine.Cli.UnitTests.ParserTests
+{
+    public partial class TabCompletionTests
+    {
+        [Fact]
+        public void Instantiate_CanSuggestTemplateOption_StartsWith()
+        {
+            ITemplateEngineHost host = TestHost.GetVirtualHost(additionalComponents: BuiltInTemplatePackagesProviderFactory.GetComponents(includeTestTemplates: false));
+            var myCommand = NewCommandFactory.Create("new", host, new TelemetryLogger(null, false), new NewCommandCallbacks());
+
+            var parseResult = myCommand.Parse("new console --framework net5.0 --l");
+            var suggestions = parseResult.GetSuggestions().ToArray();
+
+            Assert.Contains("--langVersion", suggestions);
+        }
+
+#pragma warning disable xUnit1004 // Test methods should not be skipped
+        [Fact (Skip = "doesn't work at the moment; it matches with legacy --language option which cannot be completed; to discuss how to avoid that")]
+#pragma warning restore xUnit1004 // Test methods should not be skipped
+        public void Instantiate_CanSuggestLanguages()
+        {
+            ITemplateEngineHost host = TestHost.GetVirtualHost(additionalComponents: BuiltInTemplatePackagesProviderFactory.GetComponents(includeTestTemplates: false));
+            var myCommand = NewCommandFactory.Create("new", host, new TelemetryLogger(null, false), new NewCommandCallbacks());
+
+            var parseResult = myCommand.Parse("new console --language ");
+            var suggestions = parseResult.GetSuggestions().ToArray();
+
+            Assert.Contains("C#", suggestions);
+            Assert.Contains("F#", suggestions);
+            Assert.Contains("VB", suggestions);
```

Comment:
Maybe also `Assert.DoesNotContain("net5.0", suggestions);`

Reply:
Similar cases for other tests may be added.

Reply:
added count checks